src/drm.cpp

```diff
@@ -1226,7 +1226,40 @@ int drm_prepare( struct drm_t *drm, const struct Composite_t *pComposite, const
 
 		drm->fbids_in_req.clear();
 
-		drm->pending = drm->current;
+		// If we need to modeset and we failed our preparation
```

Comment:
TBH I'd prefer to keep most of this logic inside steamcompmgr, with a `drm_rollback()` function which can be called to cancel the modeset.